### PR TITLE
Add LPRNet training code

### PR DESCRIPTION
**Pull request by alexey-trushkov**
**Before submitting a pull request**, please make sure the following is done:

1. You have merged in the latest changes from `develop`.
2. Resolved any conflicts that may have happened from that merge.
3. Ensure that you have run all tests and linting locally.

### CODE REVIEW

**Comment by alexey-sidnev** [comment]
I will take a look today.

File: training_toolbox/lpr/README.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+As training dataset for this model [Synthetic Chinese License Plates](https://download.01.org/openvinotoolkit/training_toolbox_tensorflow/datasets/Synthetic_Chinese_License_Plates.tar.gz) dataset was used.
+  
+To train a model, go through the following steps:
+
+1. Download training data and extract it in some `<dataset_folder>`. It will consist from
+    folder with training images named `crops` and text file with annotations named 
+    `annotation`. 
+    
+    After extracting training data archive run python script from
```
Comment (alexey-sidnev) [inline training_toolbox/lpr/README.md]: We have a directory with datasets: `data`. I think it is a good practice to store all datasets and processing scripts in one place.

File: training_toolbox/lpr/evaluate.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
+import random
+import numpy as np
+import tensorflow as tf
+import tensorflow.contrib.slim as slim
+import time
+from trainer import inference, LPRVocab, encode, decode_beams
+from lpr.toolbox.utils import accuracy, dataset_size
+from utils.helpers import load_module
+
+import os
```
Comment (alexey-sidnev) [inline training_toolbox/lpr/evaluate.py]: I think it is a strange order of imports. Do you have a chance to run `pylint` locally?

File: training_toolbox/lpr/infer_ie.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python
```
Comment (alexey-sidnev) [inline training_toolbox/lpr/infer_ie.py]: In SSD we don't use it, so every script is started with `python` explicitly. Could you please change it?

File: training_toolbox/lpr/toolbox/spatial_transformer_layer.py [hunk 4]
```diff
@@ -0,0 +1,203 @@
+# Copyright 2016 The TensorFlow Authors. All Rights Reserved.
```
Comment (alexey-sidnev) [inline training_toolbox/lpr/toolbox/spatial_transformer_layer.py]: Looks like it is the same file: `https://github.com/tensorflow/models/blob/master/research/transformer/spatial_transformer.py`. We use `models` repository as a submodule. Can we use the original file?

File: training_toolbox/lpr/train.py [hunk 5]
```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python
```
Comment (alexey-sidnev) [inline training_toolbox/lpr/train.py]: See the previous comment.

File: training_toolbox/lpr/infer_ie.py (hunk 3 above)
Comment (alexey-trushkov) [reply training_toolbox/lpr/infer_ie.py]: Fixed.

File: training_toolbox/lpr/evaluate.py (hunk 2 above)
Comment (alexey-trushkov) [reply training_toolbox/lpr/evaluate.py]: Fixed.

File: training_toolbox/lpr/toolbox/spatial_transformer_layer.py (hunk 4 above)
Comment (alexey-trushkov) [reply training_toolbox/lpr/toolbox/spatial_transformer_layer.py]: Fixed.

File: training_toolbox/lpr/train.py (hunk 5 above)
Comment (alexey-trushkov) [reply training_toolbox/lpr/train.py]: Fixed.

**Comment by alexey-trushkov** [comment]
@alexey-sidnev I've addressed all your comments, please proceed with the review.

**Comment by alexey-sidnev** [comment]
@alexey-trushkov , could you please merge https://github.com/opencv/training_toolbox_tensorflow/pull/25 first?

**Comment by alexey-trushkov** [comment]
@alexey-sidnev pylint checks have passed, could you please take a look,

**Comment by alexey-sidnev** [comment]
I was going to press the merge button, but there is a linting error:
```
training_toolbox/lpr/export.py:12:0: W0611: Unused dump_frozen_graph imported from utils.helpers (unused-import)
```

**Comment by alexey-trushkov** [comment]
@alexey-sidnev it's green now.

**Pull request merged**